tests/test_scope.py

```diff
@@ -0,0 +1,55 @@
+"""Unit tests for the scope module."""
+
+# python libraries
+import pytest
+
+# 3rd party libraries
+import numpy as np
+
+# own libraries
+import pysignalscope as pss
+
+def test_from_numpy():
+    """Test for the function from_numpy()."""
+    time = [0, 1, 2 * np.pi]
+    data = [2, 3, 2]
+    frequency = 20000
+
```

Comment:
Maybe one incorrect numpy array to assure, that the generateScopeObject-method is really called. It means one test of the array with valid data and one with invalid data. Idea is to assure, that each parameter is tested minimum with positive and negative test. For the detailed range test you can link to the detailed test of the generateScopeObject-method-Unittest.

Reply:
done

Reply:
i do not understand.
the generate_scope_object() method must be called, otherwise e.g. the `mode degree` test would not result in the given output data.

Once this test runs trough, we know that `generate_scope_object` is running:
```
    scope_object = pss.HandleScope.from_numpy(period_vector_t_i, mode="deg", f0=frequency)
    np.testing.assert_array_almost_equal(scope_object.channel_time, np.array(time) / 360 / frequency)
    np.testing.assert_array_equal(scope_object.channel_data, data)
```
From `generate_scope_object`, there is only a return way trough the given method.

```
        single_dataset_channel = HandleScope.generate_scope_object(period_vector_t_i[0], period_vector_t_i[1],
                                                                   channel_label=channel_label, channel_unit=channel_unit, channel_color=None,
                                                                   channel_linestyle=None, channel_source=None)

        # Log flow control
        logging.debug(f"{class_modulename} :FlCtl Amount of Data: 1")

        return single_dataset_channel
````

Reply:
i have added the test

Reply:
Great. Yes, I agree to the positive tests. Without 'generate_scope_object' you do not get the expected result. But we need to check, if the behaviour of a negative answer from 'generate_scope_object' leads to the expected behaviour of the function 'from_numpy'.